File: demo/src/main/java/com/netease/nim/demo/session/action/GuessAction.java
package com.netease.nim.demo.session.action;

import com.netease.nim.demo.R;
import com.netease.nim.demo.session.extension.GuessAttachment;
import com.netease.nimlib.sdk.msg.MessageBuilder;
import com.netease.nimlib.sdk.msg.model.IMMessage;
import com.zxn.netease.nimsdk.business.session.actions.BaseAction;

/**
 * Created by hzxuwen on 2015/6/11.
 */
public class GuessAction extends BaseAction {

    public GuessAction() {
        super(R.drawable.message_plus_guess_selector, R.string.input_panel_guess);
    }

    @Override
    public void onClick() {
        GuessAttachment attachment = new GuessAttachment();
        IMMessage message;

        message = MessageBuilder.createCustomMessage(getAccount(), getSessionType(), attachment.getValue().getDesc(), attachment);

        sendMessage(message);
    }
}

File: demo/src/main/java/com/netease/nim/demo/session/action/GuessAction.kt
package com.netease.nim.demo.session.action

import com.netease.nim.demo.R
import com.netease.nim.demo.session.extension.GuessAttachment
import com.netease.nimlib.sdk.msg.MessageBuilder
import com.zxn.netease.nimsdk.business.session.actions.BaseAction

/**
 * 发送猜拳的点击行为
 */
class GuessAction : BaseAction(R.drawable.message_plus_guess_selector, R.string.input_panel_guess) {
    override fun onClick() {
        val attachment = GuessAttachment()
        val message = MessageBuilder.createCustomMessage(
            account,
            sessionType,
            attachment.value.desc,
            attachment
        )
        sendMessage(message)
    }
}